logstreams/src/test/java/io/camunda/zeebe/logstreams/impl/log/LogStreamBatchWriterImplTest.java

```diff
@@ -0,0 +1,70 @@
+package io.camunda.zeebe.logstreams.impl.log;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import io.camunda.zeebe.dispatcher.ClaimedFragmentBatch;
+import io.camunda.zeebe.dispatcher.Dispatcher;
+import io.camunda.zeebe.util.buffer.BufferUtil;
+import org.agrona.DirectBuffer;
+import org.agrona.ExpandableArrayBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.junit.jupiter.api.Test;
+import org.mockito.invocation.InvocationOnMock;
+
+final class LogStreamBatchWriterImplTest {
+  private final Dispatcher dispatcher = mock(Dispatcher.class);
+  private final LogStreamBatchWriterImpl writer = new LogStreamBatchWriterImpl(1, dispatcher);
+
+  /**
+   * This test asserts that {@link LogStreamBatchWriterImpl#canWriteAdditionalEvent(int)} computes
```

Comment:
:eyes: Would love some alternative suggestions here. I tried to get that if we ever change how a batch is claimed, i.e. what we pass as fragment count/batch length between `tryWrite` and `canWriteAdditionalEvent`, a test would fail and notify us, ensuring we are doing the change on purpose. However, I understand this is way too coupled to the implementation...I just couldn't think of something else.